src/go/rpk/pkg/cli/topic/trim.go

```diff
@@ -0,0 +1,279 @@
+			if !noConfirm {
+				err := printDeleteRecordRequest(cmd.Context(), adm, o)
+				out.MaybeDie(err, "unable to print trimming request: %v", err)
+				fmt.Println()
+				confirmed, err := out.Confirm("Confirm trimming of the above offsets?")
+				out.MaybeDie(err, "unable to confirm: %v", err)
+				if !confirmed {
+					out.Exit("Trimming canceled.")
+				}
+				fmt.Println()
+			}
+			drr, err := adm.DeleteRecords(cmd.Context(), o)
+			out.MaybeDie(err, "unable to trim records: %v", err)
+			printDeleteRecordResponse(drr)
+		},
+	}
+
+	cmd.Flags().StringVarP(&offset, "offset", "o", "", "Offset to set the partition's low watermark (start offset) to")
+	cmd.Flags().IntSliceVarP(&partitions, "partitions", "p", nil, "Comma-separated list of partitions to trim records from (default to all)")
+	cmd.Flags().BoolVar(&all, "to-hwm", false, "Set the partition's low watermark to the partition's high watermark (trim all)")
+	cmd.Flags().StringVarP(&fromFile, "from-file", "f", "", "File of topic/partition/offset for which to trim offsets for")
+	cmd.Flags().BoolVar(&noConfirm, "no-confirm", false, "Disable confirmation prompt")
+
+	cmd.MarkFlagsMutuallyExclusive("to-hwm", "offset")
+	cmd.MarkFlagsMutuallyExclusive("from-file", "offset")
+	cmd.MarkFlagsMutuallyExclusive("from-file", "partitions")
+
+	return cmd
+}
+
+// parseOffsetArgs creates the kadm.Offsets based on the passed args via rpk
+// flags. It checks if the topic exists and defaults to all partitions in the
+// case that partitions were not provided.
+func parseOffsetArgs(ctx context.Context, adm *kadm.Client, all bool, topic, offset string, partitions []int) (kadm.Offsets, error) {
+	// First we check if the topic exists
+	listed, err := adm.ListTopics(ctx, topic)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list topic %q metadata: %v", topic, err)
+	}
+	listed.EachError(func(d kadm.TopicDetail) {
+		err = fmt.Errorf("unable to list the topic %q: %v", topic, d.Err)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// If no partitions are provided, we trim the record to the
+	// provided offset in all partitions.
+	if len(partitions) == 0 {
+		listed.EachPartition(func(pd kadm.PartitionDetail) {
+			partitions = append(partitions, int(pd.Partition))
+		})
+	}
+
+	o := make(kadm.Offsets)
+	for _, p := range partitions {
+		var parsedOffset int64
+		if all {
+			parsedOffset = -1 // -1 maps to the partition's high watermark.
+		} else {
+			parsedOffset, err = parseOffsetFlag(ctx, adm, topic, offset, p)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse offset %v: %v", offset, err)
+			}
+		}
+		o.Add(kadm.Offset{
+			Topic:     topic,
+			Partition: int32(p),
+			At:        parsedOffset,
+		})
+	}
+	return o, nil
+}
+
+// parseOffsetFile parse the given file whether it's a json, yaml or
+// space-delimited text. File content: topic | partition | offset.
+func parseOffsetFile(fs afero.Fs, file string) (kadm.Offsets, error) {
+	parsed, err := out.ParseFileArray[struct {
```

Comment:
I was thinking that if a person wanted to provide a two column file, they could -- with the two columns being partition & offset. So, if this Parse errors, parse again with a struct that does not have the topic field. The two column format would require a topic arg.

Also, if a person _does_ supply a topic arg _and_ a three-col format, the topic for each offset in the file must match the topic arg.